java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/statestore/StateStoreCommitterLogEntry.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.systemtest.drivers.statestore;
+
+import sleeper.systemtest.dsl.statestore.StateStoreCommitSummary;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StateStoreCommitterLogEntry {
+
+    private static final Pattern MESSAGE_PATTERN = Pattern.compile("Lambda started at (.+)|Lambda finished at ([^ ]+) |Applied request to table ID ([^ ]+) with type ([^ ]+) at time ([^ ]+)");
+
+    private StateStoreCommitterLogEntry() {
+    }
+
+    public static Object readEvent(String message) {
+        Matcher matcher = MESSAGE_PATTERN.matcher(message);
```

Comment:
Are all message sections optional,or are there an dependancies that should be reflected in the logic below?

e,g, if the request doesn't have startTime and/or endTime don't attempt to retrieve later sections

Reply:
There's 3 different types of message, and the pattern matches each of them separately. They're separated in the pattern with `|`. Each of the capture groups will be null unless the message type that they're part of was found. I'll try and make this clearer.

Reply:
I've added comments and some more structure to the code, thanks very much.